descwl_coadd/coadd.py

```diff
@@ -1,1236 +1,697 @@
+    TODO move interpolating BRIGHT into metadetect or mdet-lsst-sim
 
-        return data
+    Parameters
+    ----------
+    exp_or_ref: afw_image.ExposureF or DeferredDatasetHandle
+        The input exposure, possible deferred
 
-    def _make_coadd(self, *, wexps, weights, input_recorder, psf_config=None):
-        """
-        make a coadd from warp images, as well as psf coadd
-        """
+    Returns
+    -------
+    exp, noise_exp
+    """
+    if isinstance(exp_or_ref, DeferredDatasetHandle):
+        exp = exp_or_ref.get()
+    else:
+        exp = exp_or_ref
 
-        # combine stack images using mean
-        stats_flags = afw_math.stringToStatisticsProperty("MEAN")
-        stats_ctrl = afw_math.StatisticsControl()
-        stats_ctrl.setCalcErrorFromInputVariance(True)
-        badmask = afw_image.Mask.getPlaneBitMask(['EDGE'])
-        stats_ctrl.setAndMask(badmask)
-        stats_ctrl.setWeighted(True)
-
-        masked_images = [w.getMaskedImage() for w in wexps]
-        stacked_image = afw_math.statisticsStack(
-            masked_images, stats_flags, stats_ctrl, weights, 0, 0)
-
-        stacked_exp = afw_image.ExposureF(stacked_image, self.coadd_wcs)
-        # stacked_exp.setWcs(self.coadd_wcs)
-        stacked_exp.getInfo().setCoaddInputs(input_recorder.makeCoaddInputs())
-        coadd_inputs = stacked_exp.getInfo().getCoaddInputs()
-
-        # Build coadd psf
-        for wexp, weight in zip(wexps, weights):
-            input_recorder.addVisitToCoadd(coadd_inputs, wexp, weight)
-
-        if psf_config is not None:
-            coadd_psf = CoaddPsf(
-                coadd_inputs.ccds,
-                self.coadd_wcs,
-                psf_config.makeControl(),
-            )
-            stacked_exp.setPsf(coadd_psf)
+    var = exp.variance.array
+    weight = 1/var
 
-        return stacked_exp
+    flag_bright_as_sat(exp)
 
-    def _get_jac(self, *, cenx, ceny):
-        """
-        get jacobian at the coadd image center, and make
-        an ngmix jacobian with center specified (this is not the
-        position used to evaluate the jacobian)
-        """
-        import galsim
+    noise_exp, var = get_noise_exp(
+        exp=exp, rng=rng, remove_poisson=remove_poisson,
+    )
 
-        crpix = self.galsim_wcs.crpix
-        galsim_pos = galsim.PositionD(x=crpix[0], y=crpix[1])
+    # noise and image will have zeros in EDGE
+    zero_bits(
+        image=exp.image.array,
+        noise=noise_exp.image.array,
+        mask=exp.mask.array,
+        flags=afw_image.Mask.getPlaneBitMask('EDGE'),
+    )
 
-        galsim_jac = self.galsim_wcs.jacobian(image_pos=galsim_pos)
+    flags2interp = exp.mask.getPlaneBitMask(FLAGS2INTERP)
+    iimage, inoise = interpolate_image_and_noise(
+        image=exp.image.array,
+        noise=noise_exp.image.array,
+        weight=weight,
+        bmask=exp.mask.array,
+        bad_flags=flags2interp,
+    )
+    if iimage is None:
+        return None, None
```

Comment:
This should return three None's to match the return below.

Reply:
will fix